internal/utils/miscellaneous.go

```diff
@@ -62,27 +59,28 @@ func Key(namespaceOrName string, nameOpt ...string) client.ObjectKey {
 	return client.ObjectKey{Namespace: namespace, Name: name}
 }
 
-// Max returns the larger of x or y.
-func Max(x, y int) int {
-	if y > x {
-		return y
+// TypeDeref dereferences a pointer to a type if it is not nil, else it returns the default value.
+func TypeDeref[T any](val *T, defaultVal T) T {
```

Comment:
Same opinion expressed about generic function syntax being followed currently by callers of this function, that is expressed in https://github.com/gardener/etcd-druid/pull/777#discussion_r1594210522 applies here as well.

Reply:
Addressed, thanks.